lib/board.rb

```diff
@@ -0,0 +1,57 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(input)
+    @cells[input.to_i - 1]
+  end
+
+  def update(input,player)
+    @cells[input.to_i-1] = player.token
+  end
+
+  def full?
+    @cells.all? {|ele| ele != " "}
+  end
+
+  def turn_count
+    counter = 9
+    @cells.each do |ele|
+      counter -= 1 if ele == " "
+    end
+    counter
+  end
+
+  def taken?(input)
+    if (position(input) == " " || position(input) == nil)
+       false
+    else
+      true
+    end
+  end
```

Comment:
Avoid returning boolean literals from `if` statements, relying instead on the implicit return value of the boolean expression.  This also applies to `#valid_move?`.